pyvista/core/utilities/fileio.py

```diff
@@ -242,7 +242,7 @@ def read(
     if isinstance(filename, Sequence) and not isinstance(filename, str):
         multi = pyvista.MultiBlock()
         for each in filename:
-            name = Path(each).name if isinstance(each, (str, Path)) else None
+            name = Path(each).name if isinstance(each, (str, Path)) else None  # type: ignore[redundant-expr]
```

Comment:
Added because `filename` can be nested. Not in the docstring but tested.
https://github.com/pyvista/pyvista/blob/f15eff8e27c35402fa16699f7b441ab3e9178cbc/tests/core/test_utilities.py#L134-L136
I don't know how to annotate nested, I am open to suggestions/help

Reply:
can it be arbitrarily deeply nested (just like a MultiBlock nested with MultiBlock)? e.g. 
``` python
Path | str | Sequence[Path | str | Sequence[ Path | str | Sequence[...]]]
```

If so it's possible to type recursively, I think the `_SerializedDictArray` in `arrays.py` uses a recursive type, can look there for an example